### ISSUE TITLE
fix sensormodel confusion

### ISSUE THREAD

**Issue by pscholl** [pscholl/cmotion#24]
Due to name mangling, name (to be displayed to the user) and identifiers for sensors instantiation not all sensor worked. Find a way to make to this more explicit without using yet another wrapper.

**Comment by paulgavrikov** [comment]
The problem is that the human readable name might differ across platforms, while the identifier will stay the same (unless it's some odd sensor that has no default). We could solve it quickly by changing `SensorProcess.getMatchingSensor` to use the identifier instead the name.

**Comment by pscholl** [comment]
that's what the commit is for. It feels a little bit often indirected though. There are the identifiers in the discovery intents, which get copied into a list of SensorModels of which the identifier can then again be used to instantiate the actual Sensor.

 Now what happened is that the identifer got mangled to exlcude underscores. It was displayed in the list, however could not be instantiated. This can still happen when the intent contains invalid ids. However, the app will just crash then without notifying the user. The Recorder needs to be fixed here to do proper error reporting for that.

 Couldn't we create a RemoteSensor subclass of Sensor and merge SensorModel into the Sensor class?

**Comment by paulgavrikov** [comment]
Okay, sorry I haven't seen the commit until now. I'll work on the error reporting, basically I already moved the creation of the Status instance up, so it may already work.

> Couldn't we create a RemoteSensor subclass of Sensor and merge SensorModel into the Sensor class?
Not really: We'd have to use `SensorProcess.getMatchignSensor(Context c, String name)` to instantiate device on the discovering device aka the master. However the master may not have the discovered Sensor and thus won't be able to create that Object.

**Comment by pscholl** [comment]
So a sensor would not only be matched by its name but also its device?

**Comment by paulgavrikov** [comment]
No, a Sensor instance that is only available on Wear cannot be created on the Phone at all.
